### DISCUSSION TITLE
How To Preload Foreach after doing some action

### DISCUSSION THREAD

**Discussion by ezani92** [filamentphp/filament#8152] [Help]
Hi. im doing a custom page that displaying foreach item.

when i add an item using an action. looks like the foreach not refreshing automatically. i need to manual refresh the page to see the data. am i missing something? thanks for help

**Comment by leandrocfe** [comment]
Are you using something like this?
```php
protected function getViewData(): array
{
    return [
        'myItems' => Model::get()
    ];
}
```

```html
<div>
    @foreach ($myItems as $item)
        <div>{{ $item->column }}</div>
    @endforeach
</div>
```
